pkg/doc/signature/proof/jws.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package proof
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"strings"
+
+	"github.com/piprate/json-gold/ld"
+)
+
+const securityContext = "https://w3id.org/security/v2"
+
+const (
+	jwtPartsNumber   = 3
+	jwtHeaderPart    = 0
+	jwtSignaturePart = 2
+)
+
+// CreateDetachedJWTHeader creates detached JWT header.
+func CreateDetachedJWTHeader(p *Proof) string {
+	jwtHeaderMap := map[string]interface{}{
+		"alg":  p.Type,
+		"b64":  false,
+		"crit": []string{"b64"},
+	}
+
+	jwtHeaderBytes, _ := json.Marshal(jwtHeaderMap) //nolint:errcheck
```

Comment:
If you want to ignore error handling, better to use panic.

```
jwtHeaderBytes, err := json.Marshal(jwtHeaderMap)
if err != nil {
    panic(err)
}
```

Reply:
I remember @troyronda advised not to **panic** in production code. @troyronda what's the best approach here? `jwtHeaderMap` is a constant JSON document defined in the same file (`jws.go`).

Reply:
In this case, panic is better. At least you can see the problem if it happens.

Reply:
@kdimak It is true that as a library we must avoid panic.

- https://github.com/golang/go/wiki/CodeReviewComments#handle-errors
- https://golang.org/doc/effective_go.html#panic
- https://github.com/uber-go/guide/blob/master/style.md#dont-panic

For error situations that are supposed to be impossible, this is an exceptional situation - something must be seriously wrong with the execution environment (or something seriously wrong with an update to some library or code we are using).

Reply:
@troyronda thank you for the really detailed explanation! So I will put a panic() here as @soluchok suggested, thanks Andrew!

Reply:
Corrected.